### PR TITLE
examples: add gradient-boosting-regression

### PR DESCRIPTION
**Pull request by Heluwe**
Add sklearn gradient boosting regression example, see https://scikit-learn.org/stable/auto_examples/ensemble/plot_gradient_boosting_regression.html?highlight=feature_importance

### CODE REVIEW

File: example/boa/sklearn/gradient-boosting-regression.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+const pos = np.add(np.arange(sorted_idx.shape[0]), 0.5);
+
+console.log("mse", mse);
+
+const test_score = np.zeros((params['n_estimators']), boa.kwargs({ dtype: np.float64 }));
+for (const [i, y_pred] of enumerate(clf.staged_predict(X_test))) {
+    test_score[i] = clf.loss_(y_test, y_pred);
+}
+
+// 实现numpy的高级索引功能
```
Comment (yorkie) [inline example/boa/sklearn/gradient-boosting-regression.js:39]: Thank you @Heluwe May I ask you to translate to English?

File: example/boa/sklearn/gradient-boosting-regression.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+const ndReindex = (nd_array, sorted_idx) => {
+    const tmp = [];
+    for (const [i, index] of enumerate(sorted_idx)) {
+        tmp.push(nd_array[index])
+    }
+    return tmp;
+}
+
+// 绘图
```
Comment (yorkie) [inline example/boa/sklearn/gradient-boosting-regression.js:48]: Ditto

**Comment by yorkie** [comment]
And sign our CLA is also required :)

File: example/boa/sklearn/gradient-boosting-regression.js (hunk 1 above)
Comment (Heluwe) [reply example/boa/sklearn/gradient-boosting-regression.js:39]: ok
